cmd/ledger/main.go

```diff
@@ -39,16 +41,30 @@ func main() {
 		return
 	}
 
+	// Determine result json file location
+	var resultFilepath string
+	if destination != nil {
+		resultFilepath = *destination
+	} else {
+		resultFilepath, err = os.Getwd()
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+	}
+	resultFilepath = filepath.Join(resultFilepath, resultFilename)
+
+	// Command logic
 	switch command {
 
 	case compare.FullCommand():
 
-		count, err := ledger.Compare(*snapshotPath1, *snapshotPath2, resultFilename)
+		count, err := ledger.Compare(*snapshotPath1, *snapshotPath2, resultFilepath)
 		if err != nil {
 			fmt.Println(err)
 			return
 		}
-		fmt.Printf("\nSuccessfully compared snapshots. Result saved to %s. Total differences found: %v\n", resultFilename, count)
+		fmt.Printf("\nSuccessfully compared snapshots. Result saved to %s. Total differences found: %v", resultFilepath, count)
```

Comment:
I must have missed this when I merged since this branch was written before the corrections you made to the original ledger tool utility PR. Removing the newline was not my intention thanks for catching it.

Reply:
As far as adding tests for the output, that's something I had in the original iteration but ran into some issues and decided to add that in a future PR, so that's coming up as well.